### PR TITLE
Module 3 - Balance in header, activity page, flash messages.

### PR DESCRIPTION
**Pull request by sinpai**
* Added balance in header and on activity page - http://take.ms/RRGlB
* Added model test for balance checking
* Added missed flash messages in transaction controller

### CODE REVIEW

File: app/views/shared/_navbar.html.slim [hunk 1]
```diff
@@ -6,4 +6,7 @@ nav.navbar.navbar-toggleable-md.navbar-inverse.bg-primary
     #navbar-responsive-button.collapse.navbar-collapse
     .nav.navbar-nav.navbar-right
       = link_to t('.activity'), transactions_path, class: 'nav-link mr-4'
+      - if current_user
+        span.nav.navbar-text.mr-1= t('.balance')
+        span.nav.navbar-text.mr-4= number_to_currency(current_user.user_balance.to_i)
```
Comment (AleksSenkou) [inline app/views/shared/_navbar.html.slim]: @sinpai let's do `to_i` in `user_balance` method
```ruby
def user_balance
  transactions.sum(:sum).to_i
end
```

File: app/views/transactions/index.html.slim [hunk 2]
```diff
@@ -1,3 +1,7 @@
+.row
+  .col-md-10
+    h5.nav.navbar-text.mr-1= t('.balance')
+    h5.nav.navbar-text.mr-4= number_to_currency(current_user.user_balance.to_i) if current_user
```
Comment (AleksSenkou) [inline app/views/transactions/index.html.slim]: @sinpai 
```ruby
h5.nav.navbar-text.mr-1= t('.balance')
h5.nav.navbar-text.mr-4= number_to_currency(current_user.user_balance.to_i) if current_user
```
is duplicated in two placeses, what do you think about DRY partial way?

File: app/views/transactions/index.html.slim [hunk 3]
```diff
@@ -1,3 +1,7 @@
+.row
+  .col-md-10
+    h5.nav.navbar-text.mr-1= t('.balance')
```
Comment (AleksSenkou) [inline app/views/transactions/index.html.slim]: @sinpai so, if `current_user` isn't set, then `Balance:` text will be shown without number, it's a little bit confusing. What do you think about showing or not showing both `Balance:` and a number?

File: app/models/user.rb [hunk 4]
```diff
@@ -7,4 +7,8 @@ class User < ApplicationRecord
 
   has_many :transactions, dependent: :destroy
   has_many :authorizations, dependent: :destroy
+
+  def user_balance
+    transactions.sum(:sum)
```
Comment (AleksSenkou) [inline app/models/user.rb]: @sinpai 
```ruby
sum(:sum)
```
looks strange 😄 

what do you think about renaming `sum` column to `amount`? for clear view

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  before(:all) do
+    @user = create(:user)
```
Comment (AleksSenkou) [inline spec/models/user_spec.rb]: @sinpai this was mentioned in previous PRs, so be confident about things we discussed before:

image

**Review by AleksSenkou — COMMENTED** [review]
Good steps, suggestions are left

File: app/views/transactions/index.html.slim (hunk 3 above)
Comment (sinpai) [reply app/views/transactions/index.html.slim]: Yes, it's a mistake here, for sure. I will fix this.

File: app/models/user.rb [hunk 6]
```diff
@@ -7,4 +7,8 @@ class User < ApplicationRecord
 
   has_many :transactions, dependent: :destroy
   has_many :authorizations, dependent: :destroy
+
+  def user_balance
```
Comment (AleksSenkou) [inline app/models/user.rb]: @sinpai `#user_balance` is already in `User` model, no need in adding `user_`, just `#balance` 😉

**Review by AleksSenkou — APPROVED** [review]
LGTM with suggestion

File: spec/models/user_spec.rb [hunk 7]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  let(:user) { create(:user) }
+
+  before do
+    FactoryBot.create_list(:transaction, 10, user: user, amount: 10)
```
Comment (AleksSenkou) [inline spec/models/user_spec.rb]: @sinpai can't we use `create_list` without `FactoryBot`?
Like that:
```ruby
create_list(:transaction, 10, user: user, amount: 10)
```

**Review by AleksSenkou — APPROVED** [review]
LGTM

**Pull request merged**